Detections/AuditLogs/UseraddedtoPrivilgedGroups.yaml

```diff
@@ -0,0 +1,41 @@
+id: 4d94d4a9-dc96-410a-8dea-4d4d4584188b
+name: User added to Azure Active Directory Privileged Groups
+description: |
+  'This will alert when a user is added to any of the Privileged Groups.
+  For further information on AuditLogs please see https://docs.microsoft.com/azure/active-directory/reports-monitoring/reference-audit-activities.
+  For Administrator role permissions in Azure Active Directory please see https://docs.microsoft.com/azure/active-directory/users-groups-roles/directory-assign-admin-roles'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Persistence
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1098
+  - T1078
+query: |
+  let timeframe = 1h;
+  let OperationList = dynamic(["Add member to role","Add member to role in PIM requested (permanent)"]);
+  let PrivilegedGroups = dynamic(["UserAccountAdmins","PrivilegedRoleAdmins","TenantAdmins"]);
+  AuditLogs
+  | where TimeGenerated > ago(timeframe)
+  | where LoggedByService =~ "Core Directory"
+  | where Category =~ "RoleManagement"
+  | where OperationName in~ (OperationList)
+  | mv-expand TargetResources
+  | extend modifiedProperties = parse_json(TargetResources).modifiedProperties
+  | mv-expand modifiedProperties
+  | extend DisplayName = tostring(parse_json(modifiedProperties).displayName), GroupName =  trim(@'"',tostring(parse_json(modifiedProperties).newValue))
+  | extend AppId = tostring(parse_json(parse_json(InitiatedBy).app).appId), InitiatedByDisplayName = tostring(parse_json(parse_json(InitiatedBy).app).displayName), ServicePrincipalId = tostring(parse_json(parse_json(InitiatedBy).app).servicePrincipalId), ServicePrincipalName = tostring(parse_json(parse_json(InitiatedBy).app).servicePrincipalName)
+  | where DisplayName == "Role.WellKnownObjectName"
+  | where GroupName in~ (PrivilegedGroups)
+  // If you are using PIM and want to only alert for operations from non-PIM, 
+  // apply filtering as InitiatedByDisplayName != "MS-PIM"
```

Comment:
Change this slightly and add in the filter value -
// uncommment below to apply filtering for InitiatedByDisplayName != "MS-PIM"
// | where InitiatedByDisplayName != 'MS-PIM'

Reply:
based on discussion, added filtering as default to MS-PIM and then comment to remove if required.